### ISSUE TITLE
td.replaceEsm() multiple replaces not working

### ISSUE THREAD

**Issue by IamGeoffrey** [testdouble/testdouble.js#468]
### Description
I'm using testdouble with native ESM support of NodeJS and want to mock/replace multiple imports.

### Issue
When I try to replace multiple imports it only replaces the first one. The second module gets imported without the replacement.
```js
    const { default: defaultFetch, ...restFetch } = await import('node-fetch')
    await td.replaceEsm('node-fetch', { ...restFetch }, () => 'mocked default')

    const { default: defaultDependency, ...restDependency } = await import('./localDependency.mjs')
    await td.replaceEsm('./localDependency.mjs', { ...restDependency }, { ...defaultDependency, TEST: 'test' })

    const { default: moduleToTest } = await import('./moduleToTest.mjs')
```
When I log inside the `moduleToTest.mjs` I get the replaced `node-fetch` as expected.
When logging the `localDepenency.mjs` I do get the "unreplaced" module.

When I switch the order of the `td.replaceEsm` in the test file, I only see the localDepency beeing replaced.

### Environment

- `node -v`: `v14.17.1`
- `npm -v`: `6.14.13`
- `npm ls testdouble`: `testdouble@3.16.1`

**Comment by searls** [comment]
paging @giltayar!

**Comment by giltayar** [comment]
Will look into it this week.

**Comment by giltayar** [comment]
@jasonkarns Totally forgot about this, but I've checked it and it's reproducing! 

Interestingly, if you don't import the original files before mocking, then it _will_ work. Also, and even more strangely, if you import both  imports before the first `td.replaceEsm` it also works!

Looking into it now...

**Comment by searls** [comment]
Fixed by https://github.com/testdouble/quibble/pull/57

Landed in quibble@0.6.7, testdouble@3.16.4
